app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserVM.kt

```diff
@@ -28,16 +29,58 @@ class OrdersUnderUserVM(private val orderService: OrderService, private val auth
                 .observeOn(AndroidSchedulers.mainThread())
                 .doOnSubscribe {
                     progress.value = true
-                }.doFinally {
-                    progress.value = false
                 }.subscribe({
                     order.value = it
+                    val idList = ArrayList<Long>()
+                    var subQuery = ""
+                    var eventId: Long = -1
+
+                    it.forEach {
+                        it.event?.id?.let { it1 ->
+                            idList.add(it1)
+                            eventId = it1
+                            subQuery += ",{\"name\":\"id\",\"op\":\"eq\",\"val\":\"$eventId\"}"
+                        }
+                    }
```

Comment:
All of this should be encapsulated inside the method so that the entire process can be tested